src/pages/index.js

```diff
@@ -1,248 +1,248 @@
-// });
-
-// function openModal(modal) {
-//   modal.classList.add("modal_opened");
-//   document.addEventListener("keyup", handleEscButton);
-//   document.addEventListener("click", handleOverlayClose);
-// }
-
-// function closeModal(modal) {
-//   modal.classList.remove("modal_opened");
-//   document.removeEventListener("keyup", handleEscButton);
-//   document.removeEventListener("click", handleOverlayClose);
-// }
+const editProfileButton = document.querySelector(
+  "#profile__edit-button-action"
+);
 
-const newCardPopup = new PopupWithForm({
-  popupSelector: "#contentModal",
-  handleFormSubmit: handleCardFormSubmit,
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "23172f33-55e2-4e0e-a695-0bae3ab40106",
+    "Content-Type": "application/json",
+  },
 });
 
-const profileCardPopup = new PopupWithForm({
-  popupSelector: "#profileModal",
-  handleFormSubmit: handleProfileFormSubmit,
-});
+function createCard(data) {
+  return new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleCardDeleteClick,
+    handleCardLike,
+    handleCardLikeRemove
+  );
+}
 
-const contentCardPreview = new PopupWithImage({
-  popupSelector: ".js-preview-popup",
-});
+let cardSection;
 
-const cardSection = new Section({ items: cardData }, renderCard, "#section");
-cardSection.renderItems(cardData);
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection = new Section({ items: cards }, renderCard, "#section");
 
-const newUserInfo = new UserInfo("#profilename", "#profilejob");
+    function renderCard(cards) {
+      const cardElement = createCard(cards);
+      cardSection.addItem(cardElement.getView());
+    }
 
-function renderCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  // cardListEl.prepend(cardElement.getView());
-  cardSection.addItem(cardElement.getView());
-}
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
 
-function handleImageClick(cardData) {
-  console.log(cardData);
-  contentCardPreview.openModal(cardData);
-}
+const newCardPopup = new PopupWithForm({
+  popupSelector: "#contentModal",
+  handleFormSubmit: handleCardFormSubmit,
+});
 
-// ---- Profile Modal open and close event listeners -------------------------------------
+contentAddButton.addEventListener("click", () => {
+  cardFormValidator.toggleButtonState();
+  newCardPopup.openModal();
+});
 
-const editProfileButton = document.querySelector(
-  "#profile__edit-button-action"
+function handleCardFormSubmit(cardValues) {
+  const submitButton = document.getElementById("modalContentSubmit");
+  const originalButtonText = submitButton.textContent;
+  // Change button text to "Saving..."
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardValues)
+    .then((cardData) => {
+      const newCard = createCard(cardData);
+      cardSection.prependItems(newCard.getView());
```

Comment:
`.getView()` should be inside `createCard` to fully create a card not to duplicate it in 2 places